### PR TITLE
Filter/add status plug

### PR DESCRIPTION
**Pull request by RWDai**
TODO 插件获取redis_url信息方式、filter crd刷新配置方式

### CODE REVIEW

File: kernel/Cargo.toml [hunk 1]
```diff
@@ -32,7 +33,7 @@ urlencoding.workspace = true
 async-compression.workspace = true
 
 tardis = { workspace = true, features = ["future", "crypto", "tls"] }
-
```
Comment (gudaoxuri) [inline kernel/Cargo.toml:35]: poem的引入过重了，更不要把tardis的web-server模块引进来。

**Review by gudaoxuri — CHANGES_REQUESTED** [review]
一次PR中包含太多内容，需进一步讲解下。

**Comment by RWDai** [comment]
> 一次PR中包含太多内容，需进一步讲解下。

The main changes in this PR are as follows, with additional changes specified in parentheses:

1. Modified the e2e script to support WebSocket testing (including allowing httproute to accept WebSocket connections and enabling request filters within the ws protocol for the purpose of executing request filtering only, primarily to make the rewire plugin work with the ws protocol).

2. Moved SgRouteFilterContext to the plugin level and renamed it to SgRoutePluginContext, providing a broader scope of usage for the Context.

3. Filter changes: Added parameters to the init function to indicate all the routing rules that the filter can accept. Replaced the kind method in the SgPluginFilter trait with the accept method, which is used at runtime to determine whether the filter should be executed.

4. Added several filters.

============================================
本pr主要变更有以下，括号内为附带变更
1. 修改e2e脚本,支持websocket等测试(附带:httproute允许接受websocket连接,请求filter允许在ws协议内启用（仅执行请求过滤，主要为了rewire插件对ws协议生效）)
2. 把SgRouteFilterContext提到plugin层级变为SgRoutePluginContext，语义上Context可以使用范围更广。
3. filter变更：init添加参数标明过滤器所可以接受的所有路由规则；SgPluginFilter trait的kind方法替换为accept方法，用于运行时决定是否执行filter。
4. 添加若干过滤器。

File: kernel/Cargo.toml (hunk 1 above)
Comment (RWDai) [reply kernel/Cargo.toml:35]: poem引用原来是为了status插件使用，那么我会使用hpyer库直接替换

Comment (gudaoxuri) [reply kernel/Cargo.toml:35]: 是的，hyper就可以了。

**Pull request merged**